lib/pos_mempool_test.go

```diff
@@ -0,0 +1,249 @@
+package lib
+
+import (
+	"bytes"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/stretchr/testify/require"
+	"math/rand"
+	"os"
+	"testing"
+)
+
+func TestPosMempoolStart(t *testing.T) {
+	require := require.New(t)
+
+	params := DeSoTestnetParams
+	globalParams := _testGetDefaultGlobalParams()
+
+	dir, err := os.MkdirTemp("", "badgerdb-mempool")
+	require.NoError(err)
+	t.Logf("BadgerDB directory: %s\nIt should be automatically removed at the end of the test", dir)
+	defer os.RemoveAll(dir)
+
+	mempool := NewPosMempool(&params, globalParams, nil, 0, dir)
+	require.NoError(mempool.Start())
+	require.Equal(PosMempoolStatusRunning, mempool.status)
+	mempool.Stop()
+}
+
+func TestPosMempoolRestartWithTransactions(t *testing.T) {
+	require := require.New(t)
+	seed := int64(991)
+	rand := rand.New(rand.NewSource(seed))
+
+	globalParams := _testGetDefaultGlobalParams()
+	feeMin := globalParams.MinimumNetworkFeeNanosPerKB
+	feeMax := uint64(10000)
+
+	params, db := _blockchainSetup(t)
+	m0PubBytes, _, _ := Base58CheckDecode(m0Pub)
+	m1PubBytes, _, _ := Base58CheckDecode(m1Pub)
+
+	latestBlockView, err := NewUtxoView(db, params, nil, nil)
+	require.NoError(err)
+
+	dir, err := os.MkdirTemp("", "badgerdb-mempool")
+	require.NoError(err)
+	t.Logf("BadgerDB directory: %s\nIt should be automatically removed at the end of the test", dir)
+	defer os.RemoveAll(dir)
+
+	mempool := NewPosMempool(params, globalParams, latestBlockView, 2, dir)
+	require.NoError(mempool.Start())
+	require.Equal(PosMempoolStatusRunning, mempool.status)
+
+	txn1 := _generateTestTxn(t, rand, feeMin, feeMax, m0PubBytes, m0Priv, 100, 0)
+	txn2 := _generateTestTxn(t, rand, feeMin, feeMax, m1PubBytes, m1Priv, 100, 0)
+	require.NoError(mempool.AddTransaction(txn1))
+	require.NoError(mempool.AddTransaction(txn2))
+
+	poolTxns := mempool.GetTransactions()
+	require.Equal(2, len(poolTxns))
+	mempool.Stop()
+
+	newPool := NewPosMempool(params, globalParams, latestBlockView, 2, dir)
+	require.NoError(newPool.Start())
+	require.Equal(PosMempoolStatusRunning, newPool.status)
+	newPoolTxns := newPool.GetTransactions()
+	require.Equal(2, len(newPoolTxns))
+	newPool.Stop()
+}
+
+func TestPosMempoolPrune(t *testing.T) {
+	require := require.New(t)
+	seed := int64(993)
+	rand := rand.New(rand.NewSource(seed))
+
+	globalParams := _testGetDefaultGlobalParams()
+	feeMin := globalParams.MinimumNetworkFeeNanosPerKB
+	feeMax := uint64(2000)
+
+	params, db := _blockchainSetup(t)
+	m0PubBytes, _, _ := Base58CheckDecode(m0Pub)
+	m1PubBytes, _, _ := Base58CheckDecode(m1Pub)
+
+	latestBlockView, err := NewUtxoView(db, params, nil, nil)
+	require.NoError(err)
+
+	dir, err := os.MkdirTemp("", "badgerdb-mempool")
+	require.NoError(err)
+	t.Logf("BadgerDB directory: %s\nIt should be automatically removed at the end of the test", dir)
+	defer os.RemoveAll(dir)
```

Comment:
Should we move this to a helper function and use t.Cleanup? This way we can reduce code duplication across tests. Maybe even a `NewTestPoSMempool` function that takes in the appropriate attributes and then cleans itself up properly upon test completion.

Reply:
Nice! It's cool how `t.Cleanup` is like a better `defer`.